Modules/rat_rescue.py

```diff
@@ -699,30 +696,13 @@ def mark_for_deletion(self, value) -> None:
             TypeError: bad value type
             ValueError: value failed validation
         """
-        if isinstance(value, dict):
-            # checks to ensure only the required fields are present
-            if "marked" in value and "reason" in value and "reporter" in value and len(value) == 3:
-                # now sanity check the values
-                conditions = {
-                    isinstance(value["marked"], bool),
-                    isinstance(value["reason"], str) or value["reason"] is None,
-                    isinstance(value["reporter"], str) or value["reporter"] is None,
-                }
-                if all(conditions):
-                    self._mark_for_deletion = value
-
-                else:
-                    # at least one of the values was not valid
-                    raise ValueError("Data validation failed. at least one key contains invalid"
-                                     "data.")
-            else:
-                log.debug(f"Data of value is: {value}")
-                raise ValueError("required fields missing and/or keys!")
+        if isinstance(value, MarkForDeletion):
+            self._mark_for_deletion = value
         else:
-            raise TypeError(f"expected type dict, got type {type(value)}")
+            raise TypeError
```

Comment:
Could you give these some kind of message? A common convention is to simply use the name of the wrong type as a message, but anything will do.